src/journal-remote/journal-gatewayd.c

```diff
@@ -396,6 +488,21 @@ static mhd_result request_parse_arguments_iterator(
                 return MHD_YES;
         }
 
+        if (streq(key, "lines")) {
+                if (isempty(value)) {
+                        return MHD_NO;
+                }
+
+                r = safe_atoi64(value, &m->n_skip);
+                if (r < 0) {
+                        m->argument_parse_error = r;
+                        return MHD_NO;
+                }
+
+                m->n_skip *= -1;
```

Comment:
Hmm?? Why this is necessary?

Reply:
It determines [if we seek head or tail](https://github.com/systemd/systemd/pull/17314/commits/cdc2b0e58bac0baf042c1c7d8a8d9ed6f1e7713d#diff-732cab1fcb807574656bfc1b912fe5e3b41ecba76bdbda6bfb550bf3587f4681R605). When n_skip > 0, we seek on journal head, when < 0 we seek on tail. And the lines GET argument is only used for recent mode ("Show the most recent journal events and limit the number of events shown" as I put it in the doc).